### ISSUE TITLE
Removing Image Toolbar Button Causes Image Dialog to No Longer Display Width and Height Properties

### ISSUE THREAD

**Issue by HeyWrecker** [ckeditor/ckeditor4#3751]
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Added entry for `Image` to `config.removeButtons` in `config.js`.
2. Used `CKEDITOR.instances.id_message_html.setData()` method to pre-populate message window with an "HTML template"
3. Double-clicking an image in the "Message html" box displays the Image Dialog with the `txtUrl` and the `txtAlt` fields displayed along with the image preview block. However, `txtWidth` and `txtHeight` fields are missing. The ancestor `table` element of class `.cke_dialog_ui_hbox` has a `style="display: none"`.
4. This only occurs when the `Image` button is removed. However, if the `Image` button is present, all fields display.

### Expected result

Image dialog box should display all of the same properties whether the `Image` is in `config.removeButtons` or not.

### Actual result

Image dialog box hides `txtWidth` and `txtHeight` but displays `txtUrl`, and `txtAlt` as well as the image preview box.

### Note

I am not sure if the bug is that the Image Dialog loads when an image is double-clicked or right-clicked and the option "Image Properties" is clicked even though `config.removeButtons = 'Image'` is set. Or if the bug is that the `txtWidth` and `txtHeight` don't display when the button is removed.

In other words, I don't know if removing the button is supposed to remove the Image Dialog functionality and the fact that it displays is the bug or the properties not displaying is the bug.

Thanks for any help or guidance you might offer!

## Other details

* Browser: Chrome 79.x for Windows
* OS: Windows 10
* CKEditor version: 4.6.1
* Installed CKEditor plugins: Image

![image](https://user-images.githubusercontent.com/3180594/71520361-ad384c80-2889-11ea-97e5-d213169942ea.png)

**Comment by jacekbogdanski** [comment]
Works fine for me using the latest CKEditor 4.13.1 and 4.6.1 also, however, note that we are not providing backward bug fixes. Could you provide a demo with minimal CKEditor4 configuration where the issue occurs? You could use https://codepen.io, https://jsfiddle.net, https://jsbin.com or similar tool.

**Comment by HeyWrecker** [comment]
Thanks for looking into it Jacek. I will try to recreate for you in a pen environment and let you know.

**Comment by HeyWrecker** [comment]
Good morning Jacek,

The following link is a JSFiddle that recreates what I described in a very minimal way:

https://jsfiddle.net/afk4gjqb/6/

I included as few Resources as possible to try and rule any additional includes out. All of the Fiddle Resources should be stock as far as I know. The only thing that is custom is the `customConfig` value which you should be able to pull and examine from the following url:

https://spc.crystalcleardm.com/static/core/libs/ckeditor/4.6.1/config.js?t=GB8C

I hope that helps!

**Comment by jacekbogdanski** [comment]
Deleting `customConfig` indeed didn't change anything, so it seems to not cause the issue. However, once I replaced CKEditor4 CDN link https://spc.crystalcleardm.com/static/core/libs/ckeditor/4.6.1/ckeditor.js with our official CDN for this version https://cdn.ckeditor.com/4.6.1/standard/ckeditor.js it started working. It seems like you have some custom implementation of dialog, image plugin or other stuff which results in missing fields. Please, provide official CKEditor4 resources, we can't help with code we don't own. You can see the result with of replaced CDN here: https://jsfiddle.net/543qufdg/

**Comment by HeyWrecker** [comment]
That's really interesting because as you can see, the `ckeditor.js` file we've been using is still minified and  has had all of the variable names, etc... obfuscated so to the best of my knowledge it has never been changed. With that said, I'll grab a fresh copy from the CDN and try that in our local environment. Thank you very much for taking the time to help narrow down what the problem could be. It is truly appreciated Jacek! Happy New Year!

**Comment by HeyWrecker** [comment]
I downloaded the CDN version that you referenced and swapped it out with the `4.6.1` version I was using. Unfortunately, it didn't work due to differences in the build options (I diffed them and noticed some chunks that were present in my copy of `4.6.1` that weren't in the CDN).

So, I instead utilized the `4.13.1` "[full-all](https://cdn.ckeditor.com/4.13.1/full-all/ckeditor.js)" version from the CDN and the same thing occurred. I was able to recreate it in a Fiddle again:

https://jsfiddle.net/f3zujdkc/2/

The only resources included are the CDN version of the full-all `4.13.1` CKEditor 4 and the CDN version of jQuery `1.11.2`. I built the configuration inline as you can see and eventually discovered what was causing the `width` and `height` fields to not show.

I did not realize that they were linked to the `styles` attribute of the `img` tag set in the `allowedContent` object as I thought that they would be considered attributes. Maybe I did not read the documentation closely enough?

Thank you again for your time!

**Comment by jacekbogdanski** [comment]
I'm glad that it works. Upgrading to the latest `4.13.1` also should result in better stability and new features, so I can highly recommend it. Happy New Year :)

**Comment by HeyWrecker** [comment]
Jacek, did I miss that in the documentation somehow?

**Comment by jacekbogdanski** [comment]
Overall it's not explicitly wroten down in Image plugin docs, however, if read [ACF docs](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_advanced_content_filter.htm) you can learn how ACF works and why some dialog fields are disabled if not allowed by ACF. As an example see:
https://ckeditor.com/docs/ckeditor4/latest/guide/dev_advanced_content_filter.html#custom-mode-example article, especially example with listing explanation:

> Several toolbar buttons and dialog window fields that are responsible for the features which were not explicitly listed as allowed will be removed.

**Comment by HeyWrecker** [comment]
Thanks Jacek!
